recipe/meta.yaml

```
@@ -1,45 +1,59 @@
 {% set name = "cuda-python" %}
-{% set version = "11.8.1" %}
+{% set version = "12.0.0" %}
 {% set major_version = version.split(".")[0] %}
 {% set minor_version = version.split(".")[1] %}
 {% set patch_version = version.split(".")[2] %}
+{% set target_name = "x86_64-linux" %}  # [linux64]
+{% set target_name = "ppc64le-linux" %}  # [ppc64le]
+{% set target_name = "sbsa-linux" %}  # [aarch64]
+{% set target_name = "x64" %}  # [win]
 
 package:
   name: {{ name|lower }}
   version: {{ version }}
 
 source:
   url: https://github.com/NVIDIA/cuda-python/archive/refs/tags/v{{ version }}.tar.gz
-  sha256: ef54ada5d607387f59950cfb3c3852e5816b62824402d289227b8066e8353489
+  sha256: 7df0f84c4b6210112353374a7d992bfd5bba9c97358b39ea03b61bd634ce9c8a
 
 build:
-  number: 2
-  skip: true  # [py<37 or osx or cuda_compiler_version != "11.8"]
+  number: 0
+  skip: true  # [win or osx or cuda_compiler_version != "12.0"]
   script:
     - {{ PYTHON }} -m pip install . --no-deps -vv
+  script_env:
+    # Ensure that CUDA includes can be found by the host compiler
+    - CUDA_HOME="${BUILD_PREFIX}/targets/{{ target_name }}"
   run_exports:
-    - cuda-python >=11.7.1,<12  # we need this because CUDA Python 11.7.1 has ABI breaks, see conda-forge/cuda-python-feedstock#15
+    - cuda-python >=12.0.0
   ignore_run_exports:
-    - cudatoolkit
+    - {{ compiler('cuda') }}
 
 requirements:
   build:
     - {{ compiler('c') }}
     - {{ compiler('cxx') }}
     - {{ compiler('cuda') }}
-    - sysroot_linux-64 2.17  # [linux64]
+    # The packages with CUDA headers need to be added to "build" so that
+    # cuda-python can find and parse the headers. The parsing code in
+    # cuda-python doesn't support splayed layouts, i.e. packages split between
+    # "build" and "host", because $CUDA_HOME can only point to build or host.
+    - cuda-nvrtc-dev
+    - cuda-profiler-api
+    - sysroot_{{ target_platform }} 2.17  # [linux]
   host:
-    - setuptools
+    - cuda-cudart-dev
+    - cuda-nvrtc-dev
+    - cuda-profiler-api
     - cython
-    - python
-    - pyclibrary
     - pip
+    - pyclibrary
+    - python
+    - setuptools
   run:
     - python
-    - {{ pin_compatible('cudatoolkit', min_pin='x', max_pin='x') }}
+    - cuda-version >=12.0
```

Should this change with the cuda-python major.minor version?
```suggestion
    - cuda-version >={{ major_version + "." + minor_version }}
```

---

Originally the idea is that `cuda-python` would support all CUDA minor versions (for a given major version). Follow that would think this would be

```suggestion
    - cuda-version >=>={{ major_version ~ ".0" }}
```

Note: `~` treats `undefined` variables as empty strings whereas `+` errors on `undefined`. Hence the use of `~` above (to accommodate multiple parsing passes by conda-build)